src/commands/compile/internal/compileServiceFile.ts

```diff
@@ -0,0 +1,69 @@
+import path from 'path'
+import { Target, ServerType } from '@sasjs/utils/types'
+
+import { getMacroCorePath } from '../../../utils/config'
+import { createFile, readFile } from '../../../utils/file'
+import { loadDependencies } from './loadDependencies'
+import { getServerType } from './getServerType'
+
+export async function compileServiceFile(
+  target: Target,
+  filePath: string,
+  macroFolders: string[],
+  programFolders: string[]
+) {
+  let dependencies = await loadDependencies(
+    target,
+    filePath,
+    macroFolders,
+    programFolders
+  )
+
+  const serverType = await getServerType(target)
+  const preCode = await getPreCodeForServicePack(serverType)
+
+  dependencies = `${preCode}\n${dependencies}`
+
+  if (dependencies) await createFile(filePath, dependencies)
```

Comment:
`dependencies` will always be true in this case

Reply:
@saadjutt01 you've missed this one

Reply:
it was also fixed with last commit, it has _outdated_ label